cmd/k3s/main.go

```diff
@@ -205,6 +206,20 @@ func stageAndRun(dataDir, cmd string, args []string) error {
 
 	logrus.Debugf("Running %s %v", cmd, args)
 
+	if runtime.GOOS == "windows" {
+		// syscall.Exec: not supported by windows
+		if calledAsInternal {
+			args = args[1:]
+		}
+		cmd := exec.Command(cmd, args...)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		cmd.Stdin = os.Stdin
+		cmd.Env = os.Environ()
+		err := cmd.Run()
+		return err
+	}
+
 	return syscall.Exec(cmd, args, os.Environ())
```

Comment:
Move the `syscall.Exec` or `exec.Command` bits into an `exec` function in main_linux.go/main_windows.go, and call that from here. If we have OS-specific logic, I'd prefer to keep it in OS-specific files as opposed to checking the OS at runtime.